docs/reference/docs/data-replication.asciidoc

```diff
@@ -0,0 +1,136 @@
+of the nodes and may decide to pro-actively demote a primary. This typically happens when the node holding the primary
+ is isolated from the cluster by a networking issue. More on this later.
+
+Once the operation has been  successfully performed on the primary, the primary has to deal with potential failures
+when executing it on the replica shards. This may be caused by an actual failure on the replica or due to a network
+issue preventing the operation from reaching the replica (or preventing the replica from responding). All of these
+share the same end result - a replica which is part of the in-sync replica set misses an operation that is about to
+be acknowledged. In order to avoid violating the invariant, the primary sends a message to  the master requesting
+that the problematic shard be removed from the in-sync replica set. Only once this has successfully been done, does
+the primary acknowledge the operation. Note that the master will also instruct another node to start building a new
+shard copy in order to restore the system to a healthy state.
+
+While forwarding an operation to the replicas, the primary will use the replicas to validate that it is still the
+active primary. If the primary has been isolated due to a network partition (or a long GC) it may continue to process
+ incoming indexing operations before realising that it has been demoted. Operations that come from a stale primary
+ will be rejected by the replicas. The primary, following its protocol, will then reach out to the master and will
+ learn that it has been replaced. The operation is then routed to the new primary.
+
+What happens if there are no replicas? This is a valid scenario that can happen due to index configuration or simply
+because the replicas have failed. In that case the primary is processing operations without any external validation,
+which may seem problematic. On the other hand, the primary cannot fail other shards on its own. This means that the
+master knows that the primary is the only single good copy. We are therefore guaranteed that the master will not promote
+any other (out-of-date) shard copy to be a  new primary and that any operation indexed into the primary will not be lost.
+Of course, since at that point we run with only single copy of the data, physical hardware issues can cause data loss
+(see <<index-wait-for-active-shards>> for some mitigation options).
+
+=== Basic read model
+
+Reads in Elasticsearch can be very lightweight lookups by ID or a heavy search request with complex aggregations that
+take non-trivial CPU power. One of the beauties of the primary-backup model is that it keeps all shard copies identical
+(with the exception of in-flight operations).. As such, a single copy is sufficient to serve a read requests.
+
+When a read request is sent to a node, that node is responsible for forwarding it to the right places, collating the
```

Comment:
When a read request from a client is received by a node
right places -> nodes that hold the relevant shards